java/algorithm/pom.xml

```diff
@@ -50,4 +55,68 @@
 
   <build>
   </build>
+
+  <profiles>
+    <profile>
+      <id>spotless-jdk11+</id>
+      <activation>
+        <jdk>[11,]</jdk>
+      </activation>
+      <build>
+        <plugins>
+          <plugin>
+            <groupId>com.diffplug.spotless</groupId>
+            <artifactId>spotless-maven-plugin</artifactId>
+            <version>${spotless.version.jdk11}</version>
+            <configuration>
+              <formats>
+                <format>
+                  <!-- configure license for xml files -->
+                  <includes>
+                    <include>pom.xml</include>
+                  </includes>
+                  <licenseHeader>
+                    <file>${maven.multiModuleProjectDirectory}/java/spotless/asf-xml.license</file>
+                    <delimiter>(&lt;configuration|&lt;project)</delimiter>
+                  </licenseHeader>
+                </format>
+                <format>
+  <!-- configure license for java files -->
+  <includes>
+    <include>**/*.java</include>
+  </includes>
+  <licenseHeader>
+    <file>${maven.multiModuleProjectDirectory}/java/spotless/asf-java.license</file>
+    <delimiter>package</delimiter>
+  </licenseHeader>
+</format>
+              </formats>
```

Comment:
nit: is formatting incorrect here?

Reply:
Should we add a formatter for XML too? :-)

Reply:
Yes, if there's a pre-commit plugin for XML that would be nice.

Reply:
We may be able to use https://github.com/prettier/plugin-xml via Prettier.

@vibhatha Could you open a new issue for this?

Reply:
This is addressed by #41173 I believe

Reply:
cc @kou

Reply:
Can we integrate spotless with our `.pre-commit-config.yaml`?
https://github.com/apache/arrow/blob/main/.pre-commit-config.yaml

We can detect format error in our lint CI https://github.com/apache/arrow/blob/main/.github/workflows/dev.yml#L38 by integrating it with pre-commit.

Reply:
Sure @kou I will take a look.